lib/discordrb/data/message.rb

```diff
@@ -390,5 +357,32 @@ def buttons
 
       results.flatten.compact
     end
+
+    # The user who sent this message message. Will be a {Member} most of the time, a {User} in the case
+    # that the user has left the server, or a {Recipient} if the message was sent in a private channel.
+    # @return [Member, User, Recipient]
+    def author
+      author_id = @author_data['id'].to_i
+      return unless author_id
```

Comment:
This and L378 mean return value can also be nil which is not in type doc. Not sure of the circumstances when this could happen, perhaps for a deleted account?

Reply:
In the case of a deleted account there will still be a dummy user associated with it. I'm not aware of a message that can exist without an author

Reply:
Do you want me to remove the guard then?

Reply:
I think we're fine to leave it after some thought